### PR TITLE
improve Ragdoll & None Role

### PR DESCRIPTION
**Pull request by warquys**
I add a générique [FakeInfoManger.cs](https://github.com/SynapseSL/Synapse/compare/synapse3...warquys:synapse3Dev?expand=1#diff-f3c247422e4d267e63ba59d9f6b0903a28665e5c23a80ba3c9850af107c3281b).
New proprets to the [SynapseRagdoll](https://github.com/SynapseSL/Synapse/compare/synapse3...warquys:synapse3Dev?expand=1#diff-5928ffb83a5afb2d5f15943e7b2368b41264d10018c6d0d86455813ae50a056a) and implement a FakeInfoManger to manage the info of the ragdoll.

It now possible to set custom info on ragdoll and set a custom death reason for the player and ragdoll in the DeathEvent.

Note : Change the info dont change the ragdoll prefab, only the info of the ragdoll.

### CODE REVIEW

File: Synapse3.SynapseModule/Patches/PlayerPatches/DeathPatches.cs [hunk 1]
```diff
@@ -32,15 +32,21 @@ public static bool PreDeath(PlayerStats __instance, DamageHandlerBase handler)
                 attacker = Synapse.Get<PlayerService>().Players
                     .FirstOrDefault(x => x.ScpController.Scp106.PlayersInPocket.Contains(player));
 
-            var ev = new DeathEvent(player, true, attacker, damageType, damage);
+            var ev = new DeathEvent(player, true, attacker, damageType, damage, null);
             Synapse.Get<PlayerEvents>().Death.Raise(ev);
 
+            if (ev.DeathMesasge != null)
+                handler = new CustomReasonDamageHandler(ev.DeathMesasge);
+
```
Comment (GrafDimenzio) [inline Synapse3.SynapseModule/Patches/PlayerPatches/DeathPatches.cs]: You can remove this since there is a duplicate down below

Comment (warquys) [reply Synapse3.SynapseModule/Patches/PlayerPatches/DeathPatches.cs]: ok

File: Synapse3.SynapseModule/Map/Objects/FakeInfoManger.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Synapse3.SynapseModule.Map.Objects;
+
+public class FakeInfoManger<TInfo> : IJoinUpdate
+{
+    private readonly IFakableObjectInfo<TInfo> _fakableObject;
+    private readonly MirrorService _mirror;
```
Comment (GrafDimenzio) [inline Synapse3.SynapseModule/Map/Objects/FakeInfoManger.cs]: You never use _mirror therefore can you remove it

File: Synapse3.SynapseModule/Map/Objects/IFakableObjectInfo.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using Synapse3.SynapseModule.Player;
+
+namespace Synapse3.SynapseModule.Map.Objects;
+
+public interface IFakableObjectInfo<TInfo>
```
Comment (GrafDimenzio) [inline Synapse3.SynapseModule/Map/Objects/IFakableObjectInfo.cs]: Typo Fakable => Fakeable

**Comment by GrafDimenzio** [comment]
A Question to the FakeInfoManager : why is the default info private? That way can you not modify the default info however you can modify what a specific player can see (or a condition) which looks to me like a weird decision

**Comment by GrafDimenzio** [comment]
I also just tested to change the values and they will not be updated so I don't see the need for this FakeInfoManager and this entire custom system to handle the Ragdoll Info

**Comment by warquys** [comment]
I hesitated to make it public but I thought it was better to make it private. The reasons are that the default info represent the values ​​of the synapse object. But in the ragdoll no information is modifiable. But if other Synapse objects have modifiable properties then it would be possible to modify the default value by modify the instance of the default class (in this case it would be necessary to pass a class info and not a struct info). I also thought it might be a bad idea if someone would like to change the info for the players.
If you think it's a mistake to make it private, you can change it

**Pull request closed without merging**